src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -214,8 +218,25 @@ public CRAMFileReader(final InputStream stream,
     public CRAMFileReader(final File cramFile,
                           final File indexFile, final CRAMReferenceSource referenceSource,
                           final ValidationStringency validationStringency) throws IOException {
-        this(new FileInputStream(cramFile), indexFile, referenceSource, validationStringency);
+        this(new FileInputStream(cramFile),
+                wrapIndexFile(cramFile, indexFile),
```

Comment:
fixed. see the updated version.  We have extracted common initialization code from the two constructors into the 'initWithStreams' method and inlined mentioned above 'wrapIndexFile' method.